activesupport/lib/active_support/time_with_zone.rb

```diff
@@ -476,6 +476,8 @@ def get_period_and_ensure_valid_local_time(period)
       end
 
       def transfer_time_values_to_utc_constructor(time)
+        # avoid creating another Time object if possible
+        return time if time.instance_of?(Time) && time.utc?
```

Comment:
@natematykiewicz Out of curiosity, why wouldn't we want to return true for these subclasses? Sounds convenient to me